### PR TITLE
Feature/c19v4332 structured logging

### PR DESCRIPTION
**Pull request by ste-banks**

### CODE REVIEW

File: src/NBS.Appointments.Service.Core/Services/QflowService.cs [hunk 1]
```diff
@@ -57,7 +67,16 @@ public class QflowService : IQflowService
...
-            return JsonSerializer.Deserialize<SiteAvailabilityResponse[]>(responseBody);
+            try
+            {
+                return JsonSerializer.Deserialize<SiteAvailabilityResponse[]>(responseBody);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Exception thrown when attempting to deserialize qflow site availability response. Message: {@Message}, Response body: {@ResponseBody}, StatusCode: {@StatusCode}",
+                    ex.Message, responseBody, response.StatusCode);
+                throw;
```
Comment (ste-banks) [inline src/NBS.Appointments.Service.Core/Services/QflowService.cs:78]: Do we want to throw here or change the response model to return a non-successful status code to NBS? Similar to some other methods further below.

File: src/NBS.Appointments.Service.Core/Services/QflowService.cs [hunk 2]
```diff
@@ -78,7 +97,16 @@ public async Task<SiteSlotsResponse> GetSiteSlotAvailability(int siteId, DateTim
...
-            return JsonSerializer.Deserialize<SiteSlotsResponse>(responseBody);
+            try
+            {
+                return JsonSerializer.Deserialize<SiteSlotsResponse>(responseBody);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Exception thrown when attempting to deserialize qflow site slot availability response. Message: {@Message}, Response body: {@ResponseBody}, StatusCode: {@StatusCode}",
+                    ex.Message, responseBody, response.StatusCode);
+                throw;
```
Comment (ste-banks) [inline src/NBS.Appointments.Service.Core/Services/QflowService.cs:108]: Same as above.

File: src/NBS.Appointments.Service.Core/Services/QflowService.cs [hunk 3]
```diff
@@ -193,9 +230,18 @@ public async Task<IList<AppointmentResponse>> GetAllCustomerAppointments(long qf
...
-            return JsonSerializer.Deserialize<List<AppointmentResponse>>(responeBody);
+            try
+            {
+                return JsonSerializer.Deserialize<List<AppointmentResponse>>(responseBody);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Exception thrown when attempting to deserialize customer appointments response. Message: {@Message}, Response body: {@ResponseBody}, StatusCode: {@StatusCode}",
+                    ex.Message, responseBody, response.StatusCode);
+                throw;
```
Comment (ste-banks) [inline src/NBS.Appointments.Service.Core/Services/QflowService.cs:243]: Same as above.

File: src/NBS.Appointments.Service.Core/Services/QflowService.cs (hunk 1 above)
Comment (vicr1) [reply src/NBS.Appointments.Service.Core/Services/QflowService.cs:78]: Yep, should throw. This class shouldn't need to understand the context under which it was called and hence shouldn't even understand that it is part of a http web api. Callers responsibility to decide what to do

File: src/NBS.Appointments.Service.Core/Services/QflowService.cs (hunk 2 above)
Comment (vicr1) [reply src/NBS.Appointments.Service.Core/Services/QflowService.cs:108]: Same response

**Review by vicr1 — APPROVED** [review]

**Pull request merged**